DualView/src/main/java/com/microsoft/device/display/samples/contentcontext/ListFragment.kt

```diff
@@ -34,56 +37,56 @@ class ListFragment : Fragment() {
     ): View? {
         val layout = inflater.inflate(R.layout.fragment_items_list, container, false)
 
-        val recyclerView = layout.findViewById<RecyclerView>(R.id.list_items)
-
-        activity?.let {
-            if (!ScreenHelper.isDualMode(it)) {
-                selectedViewModel.selectedPosition.value = -1
-            }
+        activity?.let { activity ->
+            val recyclerView = layout.findViewById<RecyclerView>(R.id.list_items)
 
             adapterItems =
                 RestaurantAdapter(
-                    it,
-                    DataProvider.restaurants,
-                    selectedViewModel.selectedPosition.value ?: -1,
+                    activity,
+                    selectedViewModel,
                     ::onItemClick
                 )
             recyclerView.adapter = adapterItems
+            recyclerView.setHasFixedSize(true)
 
             val divider = DividerItemDecoration(context, DividerItemDecoration.VERTICAL)
-            divider.setDrawable(resources.getDrawable(R.drawable.item_divider, null))
+            ResourcesCompat.getDrawable(resources, R.drawable.item_divider, null)?.let {
+                divider.setDrawable(it)
+            }
             recyclerView.addItemDecoration(divider)
-            addSelectionObserver()
+
+            listToolbar = layout.findViewById(R.id.list_toolbar)
+            listToolbar.inflateMenu(R.menu.menu_list)
+            listToolbar.setOnMenuItemClickListener { onMenuItemSelected(it) }
         }
 
-        handleSpannedModeSelection()
+        return layout
+    }
 
-        val listToolbar = layout.findViewById<Toolbar>(R.id.list_toolbar)
-        listToolbar.inflateMenu(R.menu.menu_list)
-        listToolbar.setOnMenuItemClickListener { onMenuItemSelected(it) }
+    override fun onScreenInfoChanged(screenInfo: ScreenInfo) {
+        if (!screenInfo.isDualMode()) {
+            adapterItems?.selectItem(-1)
+        }
 
-        activity?.takeIf { ScreenHelper.isDualMode(it) }?.let {
+        if (screenInfo.isDualMode()) {
             listToolbar.menu?.findItem(R.id.action_map)?.isVisible = false
         }
+    }
 
-        return layout
+    override fun onStart() {
+        super.onStart()
+        ScreenManagerProvider.getScreenManager().addScreenInfoListener(this)
     }
 
-    private fun addSelectionObserver() {
-        selectedViewModel.selectedPosition.observe(
-            requireActivity(),
-            Observer {
-                if (it != -1) {
-                    adapterItems?.selectItem(it)
-                }
-            }
-        )
+    override fun onResume() {
+        super.onResume()
+        ScreenManagerProvider.getScreenManager().removeScreenInfoListener(this)
     }
 
     private fun onMenuItemSelected(menuItem: MenuItem): Boolean {
         return when (menuItem.itemId) {
             R.id.action_map -> {
-                startDetailsFragment(null)
+                startDetailsFragment(-1)
```

Comment:
Can we please, move this magic number to a constant with a descriptive name? (I know that it was here before, just realized about it now).